src/python/WMCore/Services/WMStats/WMStatsWriter.py

```diff
@@ -99,6 +99,38 @@ def bulkUpdateData(self, docs, existingDocs):
         self.couchDB.commit(new_edits=False)
         return
 
+    def bulkUpdateDataAnUpdateCache(self, docs, revCache, secondTry=False):
+        """
+        :param docs: docs to insert or update
+        :param existingDocs: docs existing in current
+        :return:
+        """
+        if isinstance(docs, dict):
+            docs = [docs]
+
+        notInCacheKey = []
+        notInCacheDoc = []
+        for doc in docs:
+            if doc['_id'] in revCache:
+                revList = revCache[doc['_id']].split('-')
+                # update the revision number
+                doc['_rev'] = "%s-%s" % (int(revList[0]) + 1, revList[1])
+                self.couchDB.queue(doc)
+                revCache[doc['_id']] = doc['_rev']
+            else:
+                if secondTry:
+                    doc['_rev'] = "1-123456789"
+                    self.couchDB.queue(doc)
+                    revCache[doc['_id']] = doc['_rev']
+                else:
+                    idParts = doc['_id'].split('-')
```

Comment:
Shouldn't it be `doc['_rev']...`? And what happens if it's the first time we create the document, it doesn't have a revision number yet.